modules/tensor_mechanics/src/scalarkernels/PeriodicStrain.C

```diff
@@ -0,0 +1,133 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PeriodicStrain.h"
+
+// MOOSE includes
+#include "Assembly.h"
+#include "PeriodicStrainUserObject.h"
+#include "MooseVariableScalar.h"
+#include "SystemBase.h"
```

Comment:
Just to explain why you have a "non-unity" build failure. To speed up compiles we actually group several `.C` files together and compile them altogether. This means you can get away without including header files for types you might use because they are being globbed in from other unrelated files. To solve this, just make sure that you include headers for the types that you use. 

At the same time, don't get crazy and include every header every time. That can lead to slower compiles too! Use good judgement. Is this header for sure going to be in one of the headers I'm including (e.g. `<vector>` or is this head special to this file e.g. `RankTwoTensor`?).

Reply:
You can, but you'll have to recompile everything. We need to get you on Icecream too. Do you know what that is? If not, stop into the MOOSE cube and chat with Jason (@milljm).

Reply:
And the compilation is taking longer. No, I don't, will stop by.